elastic/kibana
# [maps] fix EMS vector file manifest is loaded in Discover

Closes https://github.com/elastic/kibana/issues/157429

Choropleth lens chart `registerVisualization` is called during Discover page load. This PR removes loading of EMS files from `registerVisualization` to avoid loading EMS files on Discover page load.

There is a trade off since EMS files are needed in synchronous functions such as `getSuggestions`. Before EMS files are loaded, getSuggestions will return no suggestions.

### Test instructions
1. Load discover page with network tab open. Verify no requests are made to `vector.maps.elastic.co`
2. Create a new lens visualization. Drag `geo.src` field from sample web logs into the center. Verify region map suggestion is displayed. Click it. Verify you can edit region key field.
    <img width="600" alt="Screenshot 2024-07-30 at 12 04 58 PM" src="https://github.com/user-attachments/assets/67d74c69-1f45-4c7a-a522-23735008c6dd">
